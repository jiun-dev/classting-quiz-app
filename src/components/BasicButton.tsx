import React from 'react';
import {
  TouchableOpacity,
  Text,
  StyleSheet,
  GestureResponderEvent,
} from 'react-native';
import colors from '../constants/colors';
import spacing from '../constants/spacing';
import radius from '../constants/radius';
import fontSize from '../constants/fontSize';

interface CustomButtonProps {
  title: string;
  onPress: (event: GestureResponderEvent) => void;
}

function BasicButton({title, onPress}: CustomButtonProps) {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.main,
    paddingVertical: spacing.medium,
    paddingHorizontal: spacing.large,
    borderRadius: radius.medium,
    alignItems: 'center',
    marginVertical: spacing.small,
    fontSize: fontSize.medium,
  },
  text: {
    color: colors.white,
    fontSize: fontSize.large,
  },
});

export default BasicButton;
