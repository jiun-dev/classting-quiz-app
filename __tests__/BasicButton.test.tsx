import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import BasicButton from '../src/components/BasicButton';

describe('BasicButton 컴포넌트 테스트', () => {
  test('컴포넌트 렌더링 테스트', () => {
    const {getByText} = render(
      <BasicButton title="테스트 버튼" onPress={() => {}} />,
    );
    const buttonElement = getByText('테스트 버튼');

    expect(buttonElement).toBeDefined();
  });

  test('버튼 클릭 이벤트 테스트', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <BasicButton title="테스트 버튼" onPress={onPressMock} />,
    );
    const buttonElement = getByText('테스트 버튼');

    fireEvent.press(buttonElement);

    expect(onPressMock).toHaveBeenCalled();
  });
});
