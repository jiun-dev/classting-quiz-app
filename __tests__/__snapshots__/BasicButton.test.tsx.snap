// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicButton 컴포넌트 테스트 컴포넌트 렌더링 테스트 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4CAF50",
      "borderRadius": 16,
      "fontSize": 18,
      "marginVertical": 8,
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 22,
      }
    }
  />
</View>
`;
